theforeman.org/pipelines/lib/release.groovy

```diff
@@ -1,6 +1,19 @@
-void push_foreman_rpms(repo_type, version, distro) {
+void push_foreman_rpms(repo_type, version, distros) {
     version = version == 'develop' ? 'nightly' : version
-    push_rpms("foreman-${repo_type}-${version}", repo_type, version, distro)
+    keep_old_files = version != 'nightly'
+
+    for (distro in distros) {
+        if repo_type {
+            push_rpms("foreman-${repo_type}-${version}", repo_type, version, distro, keep_old_files)
+        } else {
+            // Main foreman repository
+            if version == 'nightly' {
+                push_rpms_direct("foreman-${foreman_version}/${distro}", "${foreman_version}/${distro}")
```

Comment:
We can align this if we can push to `releases/nightly/${distro}` instead of `nightly/${distro}`. Now `releases/nightly` is a symlink to `../nightly` but I'm leaning to swapping that around (`nightly` is a symlink to `releases/nightly`) so we can get rid of the if condition. In practice our `foreman-release` RPM already uses `yum.theforeman.org/releases/nightly` anyway.